src/Raven.Server/Commercial/LicenseHelper.cs

```diff
@@ -86,6 +88,20 @@ public License TryGetLicenseFromString(bool throwOnFailure)
             return null;
         }
 
+        internal static bool TryValidateLicenseExpirationDate(License license, out DateTime expirationDate)
+        {
+            expirationDate = DateTime.MinValue;
+
+            if (license == null)
+                return false;
+
+            var licenseStatus = LicenseManager.GetLicenseStatus(license);
+            if (licenseStatus.Expiration.HasValue)
+                expirationDate = licenseStatus.Expiration.Value;
+
+            return expirationDate >= RavenVersionAttribute.Instance.ReleaseDate;
```

Comment:
Why not use the `Expired` field here?

Reply:
That's our initial approach (I haven't altered it): we want to allow the server to start even if the license has expired, and we don't want to permit updating RavenDB to a version newer than the license expiration date.

Reply:
Ok